Jamfile

```diff
@@ -785,6 +785,57 @@ lib torrent
 
 	;
 
+
+# install rules
+
+rule generate-pkg-config ( properties * )
+{
+	import property-set ;
+	import project ;
+
+	local l = [ project.target [ project.module-name "." ] ] ;
+
+	# this is the libtorrent library target
+	local t = [ $(l).find torrent : . ] ;
+
+	# these are the properties we're using to build it with
+	local props = [ $(t).generate [ property-set.create $(properties) ] ] ;
+	props = $(props[1]) ;
+
+	local FULL_VERSION = 1.2.12 ;
+	local p = [ package.paths libtorrent : $(properties) ] ;
```

Comment:
I had a problem here with `b2 install`, here's the full build output:

```shell
$ mktemp -d

$ cd /tmp/tmp.ntTjfwZK1E

$ git clone https://github.com/arvidn/libtorrent.git
Cloning into 'libtorrent'...

$ cd libtorrent

$ git checkout v1.2.12
Note: switching to 'v1.2.12'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at e3f2b016d update revision
$ git clean --force

$ git submodule update '--depth=1' --init --recursive
Submodule 'simulation/libsimulator' (https://github.com/arvidn/libsimulator.git) registered for path 'simulation/libsimulator'
Cloning into '/tmp/tmp.ntTjfwZK1E/libtorrent/simulation/libsimulator'...
From https://github.com/arvidn/libsimulator
 * branch            6ecbc7edfca9f95ada50d58c54d3de92bfe3890b -> FETCH_HEAD
Submodule path 'simulation/libsimulator': checked out '6ecbc7edfca9f95ada50d58c54d3de92bfe3890b'

$ git apply /5874.patch

$ rm /5026.patch

$ PREFIX=/usr

$ BUILD_CONFIG='release cxxstd=14 crypto=openssl warnings=off address-model=32 -j8'

$ BOOST_ROOT= b2 release 'cxxstd=14' 'crypto=openssl' 'warnings=off' 'address-model=32' -j8 'link=shared' install '--prefix=/usr'
warning: mismatched versions of Boost.Build engine and core
warning: Boost.Build engine (b2) is 4.0.1
warning: Boost.Build core (at /usr/share/boost-build/src) is 2018.02-git
CXXFLAGS =
LDFLAGS =
OS = LINUX
warning: No toolsets are configured.
warning: Configuring default toolset "gcc".
warning: If the default is wrong, your build may not work correctly.
warning: Use the "toolset=xxxxx" option to override our guess.
warning: For more configuration options, please consult
warning: http://boost.org/boost-build2/doc/html/bbv2/advanced/configuration.html
Jamfile:807: in Jamfile</tmp/tmp.ntTjfwZK1E/libtorrent>.generate-pkg-config
ERROR: rule "package.paths" unknown in module "Jamfile</tmp/tmp.ntTjfwZK1E/libtorrent>".
/usr/share/boost-build/src/kernel/modules.jam:107: in modules.call-in
/usr/share/boost-build/src/util/indirect.jam:105: in indirect.call
/usr/share/boost-build/src/build/property.jam:144: in property.evaluate-conditionals-in-context
/usr/share/boost-build/src/build/targets.jam:1087: in evaluate-requirements
/usr/share/boost-build/src/build/targets.jam:1121: in common-properties2
/usr/share/boost-build/src/build/targets.jam:1017: in targets.common-properties
/usr/share/boost-build/src/build/targets.jam:1313: in class@basic-target.generate
/usr/share/boost-build/src/build/targets.jam:812: in generate-really
/usr/share/boost-build/src/build/targets.jam:784: in class@main-target.generate
/usr/share/boost-build/src/build/targets.jam:933: in targets.generate-from-reference
/usr/share/boost-build/src/build/targets.jam:1276: in generate-dependencies
/usr/share/boost-build/src/build/targets.jam:1333: in class@basic-target.generate
/usr/share/boost-build/src/build/targets.jam:812: in generate-really
/usr/share/boost-build/src/build/targets.jam:784: in class@main-target.generate
/usr/share/boost-build/src/build-system.jam:797: in load
/usr/share/boost-build/src/kernel/modules.jam:295: in import
/usr/share/boost-build/src/kernel/bootstrap.jam:139: in boost-build
/usr/share/boost-build/boost-build.jam:8: in module scope
```

Unsure if this PR is more intended for v2.0 going forward or if it's possible to use with v1.2 - could be a bad assumption from me.

Reply:
I suspect this is a hint of the root cause:

```
warning: mismatched versions of Boost.Build engine and core
warning: Boost.Build engine (b2) is 4.0.1
warning: Boost.Build core (at /usr/share/boost-build/src) is 2018.02-git
```

Reply:
this message suggests that the `boost-build.jam` files (that implement the build system) and the engine `b2` are from different versions. This sometimes causes errors. Are both `b2` and `/usr/share/boost-build` from the same package?

Reply:
Looks like no Alpine package has a `boost-build` file, but `b2` is v1.72.0-r6 according to https://pkgs.alpinelinux.org/package/v3.12/main/x86_64/boost.

Reply:
You're right, `boost-build` is v1.69.0-r1: https://pkgs.alpinelinux.org/package/v3.12/community/x86_64/boost-build.

Reply:
I'm actually using Alpine v3.13 now, but it looks like they haven't updated the Boost packages from v3.12, so the above are still correct.